specs/flyout/flyout.md

```diff
@@ -0,0 +1,184 @@
+
+### Use Cases
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item.
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Flyout target element. This way it can be dynamic and adapt to the available space.
+
+**Soft Dismiss (aka Light Dismiss):** - A way to quickly close the Flyout by clicking outside it, or pressing the `esc` key. An event would be emitted anytime the Flyout is dismissed. The Flyout would always be soft-dismissable. Detecting the users next click of a different element can trigger this when clicking outside the Flyout.
+
+**Focus Management:** - Should follow the same process for focus management as a Dialog, primarily as the Flyout's role will be 'dialog'.
+
+This includes:
+- Trap focus - optional, defaults to enabled - Focus is contained within the Flyout while it is visible.
+- First element focus - Focusing on the first focusable element when trap focus is enabled, but not overriding an element that is set to be focused by the author.
+- Returning focus to the target - after Flyout is closed the focus would return to the target element per the expectations of the Dialog design pattern.
+
+### Risks and Challenges
+Trap focus in the shadowDom has risk as it can be problematic and the FAST teams is currently re-evaluating how this is accomplished.
+For more insight: [Managing focus in the shadow DOM](https://nolanlawson.com/2021/02/13/managing-focus-in-the-shadow-dom/)
+
+
+### Prior Art/Examples
+
+#### Popover from Salesforce
+- https://www.lightningdesignsystem.com/components/popovers/
+
+![Popover](./images/salesforce-popover.png)
+#### Popover from Evergreen
+- triggered by a button
+- https://evergreen.segment.com/components/popover/
+
+![Popover](./images/evergreen-popover.png)
+
+#### Microsoft Windows Developer Flyout, Text Only
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+#### Microsoft Windows Developer Flyout, Actionable
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+#### Inline Dialog from ATLAssian
+- https://atlassian.design/components/inline-dialog/examples
+
+![Popover](./images/atlassian-popover.png)
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `visible` - boolean, toggle controlled by the author of whether or not the flyout is visible, defaults to undefined
```

Comment:
Did we consider logically flipping this attribute and using the platform's 'hidden' attribute instead?

Reply:
I see that Dialog uses 'hidden' and not 'visible', I'd recommend using 'hidden' here both for platform and FAST component alignment.

Reply:
Godo call